src/dynamics/variables_3D_4D.jl

```diff
@@ -0,0 +1,186 @@
+## GRID 3D
+export GridVariable3D
+
+"""
+A horizontal grid::AbstractGrid with a third dimension (time or vertical).
+As the grid is a AbstractVector (with a single index only) a GridVariable3D
+is a subtype of AbstractMatrix (and not AbstractArray{T,3}) with the horizontal
+dimension unravelled into its first dimension. To be generated and indexed like
+
+    nlat_half = 24      # resolution parameter of the horizontal grid
+    nlayer = 8          # number of vertical layers (or timesteps)
+    G = zeros(GridVariable3D, OctahedralGaussianGrid{Float32}, nlat_half, nlayer)
+    G[50, 3]            # the 50th grid point on the 3rd layer
+
+Fields are
+$(TYPEDFIELDS)"""
+struct GridVariable3D{NF, Grid<:AbstractGrid{NF}} <: AbstractMatrix{NF}
+    data::Vector{Grid}
+    npoints2D::Int
+end
+
+function Base.zeros(
+    ::Type{GridVariable3D},
+    Grid::Type{<:AbstractGrid{NF}},
+    nlat_half::Integer,
+    k::Integer,
+) where NF
+    npoints2D = RingGrids.get_npoints(Grid, nlat_half)
+    G3D = [zeros(Grid, nlat_half) for _ in 1:k]
```

Comment:
is it more convenient to have a vector of vectors rather than a two-dimensional vector?

Reply:
G3D here is a vector of vectors because the two horizontal dimensions of grid are unraveled because of our reduced grids that cannot be written as matrix

Reply:
So every grid is ::AbstractVector not Matrix even though it represents a 2D field